assembly/src/ast/mod.rs

```diff
@@ -490,10 +494,51 @@ impl Deserializable for ModuleAst {
         let num_local_procs = source.read_u16()? as usize;
         let local_procs = Deserializable::read_batch_from(source, num_local_procs)?;
 
-        match Self::new(local_procs, reexported_procs, imports, docs) {
-            Err(err) => Err(DeserializationError::UnknownError(err.message().clone())),
-            Ok(res) => Ok(res),
-        }
+        Self::new(local_procs, reexported_procs, imports, docs)
+            .map_err(|err| DeserializationError::UnknownError(err.message().clone()))
+    }
+
+    /// Returns byte representation of this [ModuleAst].
+    /// The serde options are serialized as a header to facilitate deserialization
+    pub fn to_bytes(&self, options: AstSerdeOptions) -> Vec<u8> {
+        let mut target = Vec::<u8>::default();
+
+        // serialize the options, so that deserialization knows what to do
+        options.write_into(&mut target);
+
+        self.write_into(&mut target, options);
+        target
+    }
+
+    /// Returns a [ModuleAst] struct deserialized from the provided bytes.
+    /// This function assumes that serialization was done using to_bytes(), and hence that an
+    /// [AstSerdeOptions] struct was serialized as header.
```

Comment:
formatting nit: it is usually a good idea to leave a blank line between the summary and more detailed explanation of the functionality. This way, Rust's doc formatter would be able to do a better job. In this case, this would look like so:

```
/// Returns a [ModuleAst] struct deserialized from the provided bytes.
///
/// This function assumes that serialization was done using to_bytes(), and hence that an
/// [AstSerdeOptions] struct was serialized as header.
```